docs/rfcs/0003-etag-changes.md

```diff
@@ -0,0 +1,226 @@
+
+Generating the Etag could also be combined with data to indicate versioning (i.e. create has v1 encoded, first update v2 etc.) but care must be taken to make sure that these Etags look truly random as it's possible users begin ascribing meaning to the Etag values if they are too similar or seemingly follow a convention.
+
+### Cache key
+
+If it's decided to use a cache, there are 2 potential paths that can be stored in the cache which would point at the same record, the hierarchical and flat path.  These means there are a couple of options for how to store the key in a cache.  This cache would contain the key (discussed here) and the value of the Etag.
+
+#### Store by path
+
+This is where the path to the resource would be used as the key for the cache, this would mean essentially the id on the flat path, and the public path for a hierarchical call.
+
+As the path differs by hierarchical and flat paths, this would mean essentially storing the path twice and is how the current system works.  This is simpler, but does mean that the cache key could not be in the cache for one path, when the work has been done for another path. 
+
+While there may be 2 keys added to the cache, at the point the record is retrieved, the path would be known from the `publicId` in the flat path, and the `id` from the database record when retrieving via a hierarchical path.  As such, both keys could be added to the cache at the same time meaning that the cache could be hit
+
+#### Store by id
+
+This is where the id is used regardless of if the request comes in on a hierarchical or flat path.  While this means there's more chance of hitting the cache, for a hierarchical path, the id is only known after the record has been retrieved from the database, which would break the point of caching this value.
+
+## Recommendations
+
+Based on the information discussed above, these are what the new Etags process should do:
+
+- The Etag should be stored in the database under a column on the `manifests` and `collections` table
+- While the distributed cache would be useful, as an initial implementation it would be ok to run using the direct database calls.  However, this should be monitored and if it seems that the database begins to become resource starved, then the distributed cache should be implemented.  Although, in the intervening time between deciding the cache is needed and deploying the implementation, additional resources can be dedicated to the [RDS instance](https://aws.amazon.com/rds/instance-types/) to mitigate issues with resource starvation. However, in preparation for this, an in-memory cache should be implemented using `IDistributedCache` with a short caching duration (~ 5 seconds)
```

Comment:
Can this recommendation be clarified? Starts by saying direct DB calls but finished by saving use in-memory cache.